src/infi/systray/traybar.py

```diff
@@ -122,15 +122,25 @@ def shutdown(self):
         PostMessage(self._hwnd, WM_CLOSE, 0, 0)
         self._message_loop_thread.join()
 
-    def update(self, icon=None, hover_text=None):
-        """ update icon image and/or hover text """
+    def update(self, icon=None, hover_text=None, menu_options=None):  #  "menu_options=None" added to be allow the update of the menu options
+        """ update icon image and/or hover text and/or menu options"""
         if icon:
             self._icon = icon
             self._load_icon()
         if hover_text:
             self._hover_text = hover_text
+        # "if menu_options" added to be allow the update of the menu options
+        if menu_options:
+            menu_options = menu_options or ()

```

Comment:
I don't think we need this line, since we are already checking if menu_options is not empty in the line above.

Reply:
You're right, I removed it!